proto/task.go

```diff
@@ -112,10 +112,16 @@ func (a *Agent) checkIn(ctx context.Context, tc *taskContext, command model.Plug
 	if ctx.Err() != nil {
 		return
 	}
-	tc.currentCommand = command
-	if idleTimeout != nil {
-		idleTimeout <- duration
+
+	if idleTimeout == nil {
+		return
+	}
+
+	select {
+	case idleTimeout <- duration:
```

Comment:
What's the reason for this? If `idleTimeout` is closed, does this hit the default case?

Reply:
yep, this is (a) canonical way to do non-blocking safe sends.